### PR TITLE
123-test-h5-loader

### LINKED ISSUE #123: Test + docs: dataset/loader/h5_loader
# Issue description

Expand unit tests for h5 loader written by @s-sd , and convert the current test from unittest style to pytest style.

- class H5FileLoader
    - init, get_data, get_data_ids, get_num_images, set_group_Structure, close

## Type of Issue

Please delete options that are not relevant.

- [x] Documentation update
- [x] Test request
- [x] Linting request

#### What's the expected result?
Unit tests that cover class and methods, asserting that methods return values that are expected for each method.

### PR DESCRIPTION
**Pull request by s-sd**
# Description

added unit tests for all remaining methods for the h5 loader

Fixes #123 

## Type of change
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
ran the new tests locally

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: test/unit/test_h5_loader.py [hunk 1]
```diff
@@ -63,6 +64,22 @@ def test_set_group_structure():
     assert got == expected
 
 
+def test_set_group_structure_ungrouped():
+    """
+    check if the set_group_structure method works as intended when data is
+    not grouped
+    """
+    dir_path = "data/test/h5/paired/test"
```
Comment (NMontanaBrown) [inline test/unit/test_h5_loader.py:72]: Change to ./data/test...? relative file path works locally for me vs the current solution. Will it pass travis?

File: test/unit/test_h5_loader.py [hunk 2]
```diff
@@ -129,5 +146,51 @@ def test_get_data():
...
+        raise AssertionError
+
+
+def test_init_incompatible_conditions():
+    """
+    check if the initialisation works as expected and raises an error when
+    directories to ungrouped files is given but grouped variable is set to
+    True
+    """
+    dir_path = "data/test/h5/paired/test"
```
Comment (NMontanaBrown) [inline test/unit/test_h5_loader.py:161]: Change to ./data/test...? relative file path works locally for me vs the current solution. Will it pass travis?

File: test/unit/test_h5_loader.py [hunk 3]
```diff
@@ -129,5 +146,51 @@ def test_get_data():
...
+    msg = " ".join(execinfo.value.args[0].split())
+    assert "index out of range" in msg
+
+
+def test_get_data_incompatible_args():
+    """
+    check if the get_data method works as expected and raises an error when
+    data is ungrouped but index is not an int
+    """
+    dir_path = "data/test/h5/paired/test"
```
Comment (NMontanaBrown) [inline test/unit/test_h5_loader.py:174]: Change to ./data/test...? relative file path works locally for me vs the current solution. Will it pass travis?

File: test/unit/test_h5_loader.py [hunk 4]
```diff
@@ -129,5 +146,51 @@ def test_get_data():
...
+    with pytest.raises(AssertionError):
+        loader.get_data(index)
+
+
+def test_get_data_incorrect_args():
+    """
+    check if the get_data method works as expected and raises an error when
+    an incorrect data type is fed in
+    """
+    dir_path = "data/test/h5/paired/test"
```
Comment (NMontanaBrown) [inline test/unit/test_h5_loader.py:188]: Change to ./data/test...? relative file path works locally for me vs the current solution. Will it pass travis?

**Comment by s-sd** [comment]
@NMontanaBrown I've changed the paths to "./data/test...". CI passes with both paths so up to you which one to move forward with

**Pull request merged**